lsmtool/tableio.py

```diff
@@ -668,30 +668,30 @@ def rowStr(row, metaDict):
             hasfillVal = False
 
         d = row[colKey]
-        if type(d) is np.ndarray:
-            if np.all(d == -9999):
-                if hasfillVal:
-                    dstr = ' '
-                else:
-                    dstr = fillVal
-        elif str(d).startswith('-9999'):
+        if str(d).startswith('-9999'):
             if hasfillVal:
                 dstr = ' '
             else:
-                dstr = fillVal
+                dstr = str(fillVal)
         else:
             if type(d) is np.ndarray:
-                dlist = d.tolist()
-                # Blank the value if it's equal to fill values
-                if hasfillVal and dlist == fillVal:
-                    dlist = []
-                # Remove blanked values
-                if len(dlist) > 0:
-                    while dlist[-1] == -9999:
-                        dlist.pop()
-                        if len(dlist) == 0:
-                            break
-                dstr = str(dlist)
+                if np.all(d == -9999):
+                    if hasfillVal:
+                        dstr = ' '
+                    else:
+                        dstr = str(fillVal)
+                else:
+                    dlist = d.tolist()
+                    # Blank the value if it's equal to fill values
+                    if hasfillVal and dlist == fillVal:
+                        dlist = []
+                    # Remove blanked values
+                    if len(dlist) > 0:
+                        while dlist[-1] == -9999:
+                            dlist.pop()
+                            if len(dlist) == 0:
+                                break
+                    dstr = str(dlist)
```

Comment:
It's not the most beautiful piece of code I've ever seen :)
I get the feeling that this whole magic with `-9999` is quite awkward. Is this only for backward compatibility? The code could be so much more compact and easier to read without it.

Reply:
Yes, agreed, but I think it would take some effort to refactor it unfortunately. So I've just stuck to patching it up for now.